controllers/helmchart_controller.go

```diff
@@ -389,18 +392,50 @@ func (r *HelmChartReconciler) reconcileFromHelmRepository(ctx context.Context,
 			return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
 		}
 
-		// Find override file and retrieve contents
-		var valuesData []byte
-		cfn := filepath.Clean(chart.Spec.ValuesFile)
-		for _, f := range helmChart.Files {
-			if f.Name == cfn {
-				valuesData = f.Data
-				break
+		// Prepend a list of valuesFiles with the deprecated ValuesFile key.
+		if chart.Spec.ValuesFile != "" {
+			valuesFiles = append(valuesFiles, chart.Spec.ValuesFile)
+		}
+		valuesFiles = append(valuesFiles, chart.Spec.ValuesFiles...)
+
+		for _, v := range valuesFiles {
+			srcPath, err := securejoin.SecureJoin(tmpDir, v)
```

Comment:
Perhaps we should be using the raw data directly from `helmChart.Files` in the `HelmRepository` case?

Reply:
It would probably be better indeed to scan the `helmChart.Files` for the defined paths, and then merge all found files in the right order at once.

Although there _may_ be a reason we moved away from looking at the `helmChart.Files`, @relu possibly knows more about this, and if not, I can confirm it by looking at the previous PRs touching this subject.

Reply:
Yes, reading from `helmChart.Files` is actually the only way to deal with these override values files in the context of dealing with a `HelmRepository` sourced chart, since we never decompress the chart tarball onto the filesystem but rather just load it directly in memory (see `helmChart, err := loader.LoadArchive(res)` above).

You refer to `tmpDir` in this situation but that is not going to work as the path is actually just an empty string at this stage, it is only initialized later on and only in the situation when we need to repackage the chart.